src/plugins/discover/public/application/doc_views/doc_views_types.ts

```diff
@@ -49,17 +49,34 @@ export type DocViewRenderFn = (
   renderProps: DocViewRenderProps
 ) => () => void;
 
-export interface DocViewInput {
-  component?: DocViewerComponent;
-  directive?: AngularDirective;
+export interface BaseDocViewInput {
   order: number;
-  render?: DocViewRenderFn;
   shouldShow?: (hit: ElasticSearchHit) => boolean;
   title: string;
 }
 
-export interface DocView extends DocViewInput {
-  shouldShow: (hit: ElasticSearchHit) => boolean;
+export interface RenderDocViewInput extends BaseDocViewInput {
```

Comment:
:information_source: I've actually split those types into different types and unioned them, so the type correctly reflects, that we only expect either of those to be set.